### ISSUE TITLE
OrientGraphNoTx commits updates to OrientElement even if OValidationException is thrown

### ISSUE THREAD

**Issue by shooit** [orientechnologies/orientdb#7145]
## Expected behavior  
When using a non-transactional graph, an exception should be thrown when you try to update an element in a way that would violate a constraint, and the update should not commit.

## Actual behavior  
The exception is thrown but the update commits and there is no way to roll back. 

## Steps to reproduce  
```
@Test
public void testNoTxGraphConstraints() {
    String location = "memory:test_noTxConstraints";
    OrientGraphNoTx graphNoTx = new OrientGraphNoTx(location);
    OrientVertexType testType = graphNoTx.createVertexType("Test");
    OProperty age= testType.createProperty("age", OType.INTEGER).setMax("3");
    OrientVertex vert1 = graphNoTx.addVertex("class:Test", "age", 2);

    try {
        vert1.setProperty("age", 4);
    } catch (OValidationException e) {
        Assert.assertEquals(vert1.getProperty("age"), 2); //this fails
    } finally {
        graphNoTx.drop();
    }
}
```

**Comment by tglman** [comment]
hi @shooit,

I reproduced it and fixed, also included the test case in our suite, it will be released with 2.2.17

Regards

**Comment by shooit** [comment]
@tglman awesome, thank you. looks good to me!
